internal/emulator/services.go

```diff
@@ -5,19 +5,36 @@ import (
 	"log"
 	"testing"
 
+	"github.com/docker/go-connections/nat"
 	"github.com/fullstorydev/emulators/storage/gcsemu"
+	"github.com/testcontainers/testcontainers-go"
 	"github.com/testcontainers/testcontainers-go/wait"
 	azblob "gocloud.dev/blob/azureblob"
 )
 
 func AzureBlob(t *testing.T, bucket string) string {
 	t.Helper()
 
-	host := TestContainer(t, Container{
-		Image:   "mcr.microsoft.com/azure-storage/azurite:latest",
-		Port:    10000,
-		Command: []string{"azurite-blob", "--blobHost", "0.0.0.0"},
+	ctx := context.Background()
+	c, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image:        "mcr.microsoft.com/azure-storage/azurite:latest",
+			ExposedPorts: []string{"10000"},
+			Cmd:          []string{"azurite-blob", "--blobHost", "0.0.0.0"},
+			WaitingFor:   wait.ForListeningPort(nat.Port("10000")),
+		},
+		Started: true,
+		Logger:  NopLogger{},
 	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() { _ = c.Terminate(ctx) })
+
+	host, err := c.PortEndpoint(ctx, nat.Port("10000"), "")
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Create thin (unexported) abstraction layer which takes a `testcontainers.GenericContainerRequest` and returns `PortEndpoint` of first exposed port.